src/python/pants/backend/python/python_requirements.py

```diff
@@ -58,9 +58,16 @@ def __call__(self, requirements_relpath='requirements.txt'):
                   raise ValueError('Only 1 --find-links url is supported per requirements file')
                 repository = value
 
+    requirements_file_target_name = requirements_relpath
+    self._parse_context.create_object('files',
```

Comment:
Not sure how likely collisions are, but it would be good to make this name a bit more synthetic, and include the owning target's name as well... `'__{}_files'.format(requirements_file_target_name)`?

Reply:
Saw this comment late. Right now the mainline case is a target named `requirements.txt` (or `foo.bob` if their requirements file is `foo.bob`). Having that filename match a requirement name or hand-written target name seemed pretty unlikely to me. I can circle back with a follow-up PR if you think my unlikely guess underestimates things.

Reply:
Should be fine, unless we can imagine a case with multiple target owners of the same requirements files.

Reply:
That's not possible even prior to this change since the requirements would each be exported as targets named by their requirement key leading to a duo target for each of those.